cpp/include/rmm/device_scalar.hpp

```diff
@@ -161,7 +201,16 @@ class device_scalar {
    */
   [[nodiscard]] value_type value(cuda_stream_view stream) const
   {
-    return _storage.front_element(stream);
+    if (_host_bounce_buffer.has_value()) {
+      // Case: Copying with pinned host memory
+      RMM_CUDA_TRY(cudaMemcpyAsync(
+        *_host_bounce_buffer, data(), sizeof(T), cudaMemcpyDefault, stream.value()));
```

Comment:
I think this is what Mark meant.
```suggestion
        _host_bounce_buffer.value(), data(), sizeof(T), cudaMemcpyDefault, stream.value()));
```

Reply:
I get it!